### PR TITLE
fix: smoothing lux meter data plot

### LINKED ISSUE #1170: Luxmeter graphs are not smooth
**Actual Behaviour**

In the Luxmeter instrument, the resulting graph is full of spikes.

**Expected Behaviour**

The graph needs to be a continuous, smooth function. Current one goes to zero frequently.
PS: This was made priority high as this is need to be fixed for data logging to have values that make sense.

**Steps to reproduce it**

Observe Luxmeter instrument graphs

**LogCat for the issue**

> Not available

**Screenshots of the issue**

device-2018-07-02-063950

**Would you like to work on the issue?**

@Vikum94

### PR DESCRIPTION
**Pull request by Vikum94**
Fixes #1170 

**Changes**: 
- Smooth plot according to the given update period. 
- Chart shows full history rather than showing only 10 data points 

**Screenshot/s for the changes**: 
screenshot_2018-07-06-09-59-43

**Checklist**: 
- [ ] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [ ] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [ ] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**: 
[lux_meter_smoothing.zip](https://github.com/fossasia/pslab-android/files/2169030/lux_meter_smoothing.zip)

### CODE REVIEW

File: app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentData.java [hunk 1]
```diff
@@ -239,11 +240,11 @@ protected void onPostExecute(Void aVoid) {
         @Override
         public void onSensorChanged(SensorEvent event) {
             data = Float.valueOf(df.format(event.values[0]));
-            visualizeData();
             unRegisterListener();
         }
 
         private void visualizeData() {
+            System.out.println("Data value: "+data);
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentData.java]: Please remove this line.

File: app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentData.java [hunk 2]
```diff
@@ -239,7 +240,6 @@ protected void onPostExecute(Void aVoid) {
         @Override
         public void onSensorChanged(SensorEvent event) {
             data = Float.valueOf(df.format(event.values[0]));
-            visualizeData();
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentData.java]: Where is this method called now?

**Comment by CloudyPadmal** [comment]
The plot is always blank for me as readings are always 0.0 @abhinavraj23 @Avjeet @harsh-2711 @yatri1609 can you test this APK and update details here?

**Comment by harsh-2711** [comment]
@CloudyPadmal Same problems persists for me.

**Comment by CloudyPadmal** [comment]
@harsh-2711 can you make a fix for this issue as Vikum is not available?

**Comment by harsh-2711** [comment]
Sure @CloudyPadmal

**Comment by harsh-2711** [comment]
@CloudyPadmal Please close this PR as I have made another PR of mine solving issue #1170

**Comment by cweitat** [comment]
PR closed due to similar PR existing #1170

**Pull request closed without merging**